test/test_pynisher.py

```diff
@@ -322,45 +343,51 @@ def test_liblinear_svc(self):
         time.sleep(1)
         p = psutil.Process()
         self.assertEqual(len(p.children(recursive=True)), expected_children[context])
-        self.assertEqual(logger_mock.debug.call_count, 2)
-        self.assertEqual(logger_mock.debug.call_args_list[0][0][0],
+        # Using a picklable-logger to capture all messages
+        self.assertEqual(self.logger.debug.call_count, 4)
+        self.assertEqual(self.logger.debug.call_args_list[0][0][0],
+                         'Restricting your function to 2 seconds cpu time.')
+        self.assertEqual(self.logger.debug.call_args_list[1][0][0],
+                         'Allowing a grace period of 1 seconds.')
+        self.assertEqual(self.logger.debug.call_args_list[2][0][0],
                          'Function called with argument: (16384, 10000), {}')
-        self.assertEqual(logger_mock.debug.call_args_list[1][0][0],
+        self.assertEqual(self.logger.debug.call_args_list[3][0][0],
                          'Your function call closed the pipe prematurely -> '
                          'Subprocess probably got an uncatchable signal.')
         # self.assertEqual(wrapped_function.exit_status, pynisher.CpuTimeoutException)
-        self.assertGreater(duration, time_limit - 0.1)
-        self.assertLess(duration, time_limit + grace_period + 0.1)
-        self.assertEqual(wrapped_function.exitcode, -9)
+        if context == 'fork':
+            self.assertGreater(duration, time_limit - 0.1)
+            self.assertLess(duration, time_limit + grace_period + 0.1)
+            self.assertEqual(wrapped_function.exitcode, -9)
 
     @unittest.skipIf(not all_tests, "skipping nested pynisher test")
     def test_nesting(self):
 
-        tl = 2  # time limit
-        gp = 1  # grace period
+        tl = 3  # time limit
+        gp = 2  # grace period
 
         start = time.time()
-        nested_pynisher(level=2, cputime=2, walltime=2, memlimit=None, increment=1, grace_period=gp)
+        nested_pynisher(level=2, cputime=2, walltime=2, memlimit=None, increment=1, grace_period=gp, logger=self.logger)
         duration = time.time() - start
         print(duration)
 
         time.sleep(1)
         p = psutil.Process()
         self.assertEqual(len(p.children(recursive=True)), expected_children[context])
-        self.assertGreater(duration, tl - 0.1)
-        self.assertLess(duration, tl + gp + 0.1)
+        if context == 'fork':
```

Comment:
This is the comment we discussed offline. For forkserver and spawn, the  time is not honored, and seems like the function is prematurely closed (I believe the pre-kill signal is not reaching nicely the nested job who gets killed by the -9 signal). Do you have a suggestion on how to approach this for fork server/spawn?

Reply:
I cannot reproduce this on my machine, the unit test appears to work well with `context == 'fork'`. It's a pity that github actions doesn't work yet. Could you maybe set up github actions in a separate PR so that it runs for this one?

Reply:
You mean `context != 'fork'`?

Reply:
Yes, correct.

Reply:
I will do a push with this removed and we can see this behavior in github actions, and proceed accordingly.